simple_certmanager/utils.py

```diff
@@ -5,20 +5,65 @@
 from django.utils.encoding import force_str
 
 from cryptography import x509
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric.types import PrivateKeyTypes
 from cryptography.hazmat.primitives.serialization import load_pem_private_key
 
 logger = logging.getLogger(__name__)
 
 
-def load_pem_x509_private_key(data: bytes):
+class PrivateKeyError(Exception):
+    pass
+
+
+class KeyIsEncrypted(PrivateKeyError):
+    pass
+
+
+class BadPassword(PrivateKeyError):
+    pass
+
+
+def load_pem_x509_private_key(data: bytes, password: str | None = None):
     """
     Small wrapper around the ``cryptography.hazmat`` private key loader.
 
     Nothing in this code is really specific to x509, but the function name expresses
     our *intent* to only deal with x509 certificate/key pairs.
     """
-    # passwords are not supported
-    return load_pem_private_key(data, password=None)
+    _password = password.encode("utf-8") if password is not None else None
+    try:
+        return load_pem_private_key(data, password=_password)
+    except TypeError as exc:
+        if (
+            password is None
+            and exc.args
+            and (err := exc.args[0])
+            and "Password was not given but private key is encrypted" in err
```

Comment:
Super cool that cryptography uses Rust under the hood... but them not raising specific exceptions and us having to rely on these error strings is NOT my preferred approach :/